### PR TITLE
🎨 Log command groups

### PR DESCRIPTION
**Pull request by ryanabraham1**
Added back the global command logger including non subsystem commands. 
Name command groups with the commands that are held within them.

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -46,6 +49,33 @@ public void robotInit() {
     // Instantiate our RobotContainer. This will perform all our button bindings,
     // and put our autonomous chooser on the dashboard.
     m_robotContainer = new RobotContainer();
+
+    Map<String, Integer> commandCounts = new HashMap<>();
+    BiConsumer<Command, Boolean> logCommandFunction =
```
Comment (ThatXliner) [inline src/main/java/frc/robot/Robot.java:54]: Is this just poor man's profiling?

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -46,6 +49,33 @@ public void robotInit() {
...
+        (Command command, Boolean active) -> {
+          String name = command.getName();
+          int count = commandCounts.getOrDefault(name, 0) + (active ? 1 : -1);
+          commandCounts.put(name, count);
+
+          Logger.recordOutput(
+              "CommandsUnique/" + name + "_" + Integer.toHexString(command.hashCode()), active);
+          Logger.recordOutput("CommandsAll/" + name, count > 0);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/Robot.java:62]: Why not log the count by itself

**Review by realjoshuau — APPROVED** [review]
LGTM

File: src/main/java/frc/robot/utils/NamedCommands.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            "Select from "
+                + Arrays.toString(commands.values().stream().map(Command::getName).toArray()));
+  }
+
+  public static Command defer(Supplier<Command> supplier, Set<Subsystem> requirements) {
+    return new DeferredCommand(supplier, requirements)
+        .withName("Defer " + supplier.get().getName());
+  }
+
+  public static Command deferredProxy(Supplier<Command> supplier) {
```
Comment (ThatXliner) [inline src/main/java/frc/robot/utils/NamedCommands.java:58]: Why the name? Why not "asProxy"

File: src/main/java/frc/robot/utils/NamedCommands.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    return new ProxyCommand(supplier).withName(supplier.get().getName());
+  }
+
+  public static Command sequence(Command... commands) {
+    return new SequentialCommandGroup(commands)
+        .withName(
+            "Sequence " + Arrays.toString(Arrays.stream(commands).map(Command::getName).toArray()));
+  }
+
+  public static Command repeatingSequence(Command... commands) {
```
Comment (ThatXliner) [inline src/main/java/frc/robot/utils/NamedCommands.java:68]: Does .withName have to be the last method called??

File: src/main/java/frc/robot/utils/NamedCommands.java (hunk 3 above)
Comment (ryanabraham1) [reply src/main/java/frc/robot/utils/NamedCommands.java:58]: its from wpilib's command class, I didn't name any of these myself

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (ryanabraham1) [reply src/main/java/frc/robot/Robot.java:62]: Idk this is josh's code from last bot I didn't bother rewriting it

File: src/main/java/frc/robot/utils/NamedCommands.java (hunk 4 above)
Comment (ryanabraham1) [reply src/main/java/frc/robot/utils/NamedCommands.java:68]: I don't think so

**Pull request merged**